docs/Style.thy

```diff
@@ -219,6 +219,102 @@ lemma test_lemma3:
          case_tac h; simp)
   done
 
+section \<open>Right vs left operator-wrapping\<close>
+
+text \<open>
+  When a term is too long, there is a general consensus to wrap it at operators. However, consensus
+  has never emerged on whether the operator should then end up at the end of the line (right
+  operator wrapping), or start of the next one (left operator wrapping).
+  Mathematicians have a tendency towards right-wrapping, programmers towards left-wrapping, they
+  each have advantages in specific contexts, thus both appear in the l4v proofs and are permitted
+  style.\<close>
+
+term \<open>A \<and> B \<longrightarrow> C \<and> D\<close> \<comment> \<open>no wrapping\<close>
+
+term \<open>A \<and>
+      B \<longrightarrow>
+        C \<and>
+        D\<close> \<comment> \<open>right-wrapping\<close>
```

Comment:
Might be a bit confusing, because you'd only end up with that if all of `A`, `B`, `C`, `D` are very large. Otherwise, you'd try to keep multiple things on one line even if others need to line-break. (This is true in both styles, but it looks stranger in right-style to me)

Reply:
Maybe just adding a comment that this form would be used if all of them are large terms would avoid the confusion already.

Reply:
Tried to add some commentary here.